tests/v2/validation/provisioning/rke1/provisioning_node_driver_test.go

```diff
@@ -192,78 +162,57 @@ func (r *RKE1NodeDriverProvisioningTestSuite) ProvisioningRKE1ClusterDynamicInpu
+func (r *RKE1NodeDriverProvisioningTestSuite) testProvisioningRKE1Cluster(client *rancher.Client, provider Provider, nodesAndRoles []nodepools.NodeRoles, kubeVersion, cni string) {
+	clusterName := provisioning.AppendRandomString(provider.Name)
 
-					opts := metav1.ListOptions{
-						FieldSelector:  "metadata.name=" + clusterResp.ID,
-						TimeoutSeconds: &defaults.WatchTimeoutSeconds,
-					}
-					watchInterface, err := r.client.GetManagementWatchInterface(management.ClusterType, opts)
-					require.NoError(r.T(), err)
+	cluster := clusters.NewRKE1ClusterConfig(clusterName, cni, kubeVersion, client)
+	clusterResp, err := clusters.CreateRKE1Cluster(client, cluster)
+	require.NoError(r.T(), err)
 
-					checkFunc := clusters.IsHostedProvisioningClusterReady
+	nodeTemplateResp, err := provider.NodeTemplateFunc(client)
+	require.NoError(r.T(), err)
 
-					err = wait.WatchWait(watchInterface, checkFunc)
-					require.NoError(r.T(), err)
-					assert.Equal(r.T(), clusterName, clusterResp.Name)
-					assert.Equal(r.T(), nodePoolName, nodePool.Name)
+	nodePool, err := nodepools.NodePoolSetup(client, nodesAndRoles, clusterResp.ID, nodeTemplateResp.ID)
+	require.NoError(r.T(), err)
 
-					clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
-					require.NoError(r.T(), err)
-					assert.NotEmpty(r.T(), clusterToken)
+	nodePoolName := nodePool.Name
 
-					err = nodepools.ScaleWorkerNodePool(testSessionClient, nodesAndRoles, clusterResp.ID, nodeTemplateResp.ID)
-					require.NoError(r.T(), err)
-				})
-			}
-		}
+	opts := metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + clusterResp.ID,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
 	}
-}
+	watchInterface, err := r.client.GetManagementWatchInterface(management.ClusterType, opts)
+	require.NoError(r.T(), err)
 
-func (r *RKE1NodeDriverProvisioningTestSuite) TestProvisioning() {
-	for _, providerName := range r.providers {
-		provider := CreateProvider(providerName)
-		r.ProvisioningRKE1Cluster(provider)
-	}
-}
+	checkFunc := clusters.IsHostedProvisioningClusterReady
 
-func (r *RKE1NodeDriverProvisioningTestSuite) TestProvisioningDynamicInput() {
-	clustersConfig := new(provisioning.Config)
-	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
-	nodesAndRoles := clustersConfig.NodesAndRolesRKE1
+	err = wait.WatchWait(watchInterface, checkFunc)
+	require.NoError(r.T(), err)
+	assert.Equal(r.T(), clusterName, clusterResp.Name)
+	assert.Equal(r.T(), nodePoolName, nodePool.Name)
 
-	if len(nodesAndRoles) == 0 {
-		r.T().Skip()
-	}
+	clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
+	require.NoError(r.T(), err)
+	assert.NotEmpty(r.T(), clusterToken)
 
-	for _, providerName := range r.providers {
-		provider := CreateProvider(providerName)
-		r.ProvisioningRKE1ClusterDynamicInput(provider, nodesAndRoles)
-	}
+	err = nodepools.ScaleWorkerNodePool(client, nodesAndRoles, clusterResp.ID, nodeTemplateResp.ID)
```

Comment:
Yeah I thought of doing that initially as well. I didn't here because I'd have to make at least three new functions: a static function that looks for a private scaling method, a dynamic function that looks for a private scaling method and the actual private method itself.

I don't mind doing that if we think that's best, I just didn't initially.

Reply:
I am flexible to what the team thinks, but I personally think it should be a separate test case, similar to what was done in the host provider tests. The scaling adds a lot of test time, to a test where maybe we just we want to validate a specific thing/things rather than waiting for it be scaled.

Reply:
that's a good point. I can add a new test case for it no worries! I'll try to see if there's a way to reduce redundancy since the scale test needs to provision a cluster first...I'll tinker around and see what I can do.

Reply:
Resolved this by adding new `r.Run` to both static and dynamic functions. By default, both the provisioning and scale functions will run. We can use regex to dictate only one function runs.

To make it easier, I added `scaleName` variable to assist making the regex easier. All that is needed is adding `scaling` at the end of the name, as seen in the code.